mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainClusterController.java

```diff
@@ -202,11 +214,11 @@ private void createOrUpdateKafkaBroker(ServiceDomainCluster sdc) {
         final Kafka currentKafka = client.resources(Kafka.class).inNamespace(client.getNamespace()).withName(sdcName).get();
 
         if (currentKafka == null) {
-            client.resources(Kafka.class).create(desiredKafka);
+            client.resources(Kafka.class).inNamespace(client.getNamespace()).create(desiredKafka);
             LOGGER.debug("{} kafka broker was missing, creating it", sdcName);
         } else {
-            if (!Objects.equals(currentKafka.getSpec(), desiredKafka.getSpec())) {
-                client.resources(Kafka.class).replace(desiredKafka);
+            if (!Objects.equals(currentKafka.getSpec(), desiredKafka.getSpec()) || !Objects.equals(currentKafka.getMetadata().getOwnerReferences(), desiredKafka.getMetadata().getOwnerReferences())) {
```

Comment:
You shouldn't compare the ownerReferences in any case. Focus on the spec

Reply:
Done